msgpack/shared/src/test/scala/fs2/data/msgpack/RoundtripTest.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2024 fs2-data Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2
+package data.msgpack
+
+import high._
+
+import cats.effect.IO
+import cats.Show
+import org.scalacheck.Arbitrary
+import weaver._
+import weaver.scalacheck._
+
+import cats.syntax.traverse._
+import org.scalacheck.Gen
+
+object RoundtripTest extends SimpleIOSuite with Checkers {
+  // With implicit generator
+  def ride[A: Show: Arbitrary: MsgpackSerializer: MsgpackDeserializer]: IO[Expectations] =
```

Comment:
Suggestion: Instead of having the method body twice, delegate from here to `ride(gen: Gen[A])` using `implicitly[Arbitrary[A]].arbitrary`. Keeps the code DRY.

Nitpick: `onlyOrError` instead of `lastOrError`.